common/rpc/params_test.go

```diff
@@ -45,6 +47,9 @@ func TestNewParams(t *testing.T) {
 	_, err = NewParams(serviceName, makeConfig(config.Service{RPC: config.RPC{BindOnIP: "invalidIP"}}))
 	assert.EqualError(t, err, "failed to get listen IP: unable to parse bindOnIP value or it is not an IPv4 address: invalidIP")
 
+	_, err = NewParams(serviceName, &config.Config{Services: map[string]config.Service{"frontend": {}}})
+	assert.EqualError(t, err, "failed to create public client outbound: need to provide an endpoint config for PublicClient")
```

Comment:
nit: error messages can be shorter. Ref: https://github.com/uber-go/guide/blob/master/style.md#error-wrapping

Reply:
Agree, will update.